Weather/ViewModels/CityViewModel.swift

```diff
@@ -0,0 +1,204 @@
+//
+//  CityViewModel.swift
+//  Weather
+//
+//  Created by Alex on 31.07.22.
+//
+
+import Foundation
+import SwiftUI
+import CoreLocation
+
+class CityViewModel: ObservableObject {
+    
+    @Published var weather = WeatherResponse.emptyResponse()
+    @Published var city: String = "Minsk" {
+        didSet {
+            getLocation()
+        }
+    }
+    
+    private lazy var dateFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateStyle = .full
+        return formatter
+    }()
+    
+    private lazy var dayFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "EEE"
+        return formatter
+    }()
+    
+    private lazy var timeFormatter: DateFormatter = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "hh a"
+        return formatter
+    }()
+    
+    init() {
+        getLocation()
+    }
+    
+    var date: String {
+        return dateFormatter.string(from: Date(timeIntervalSince1970: TimeInterval(weather.current.dt)))
+    }
+    
+    var weatherIcon: String {
+        if weather.current.weather.count > 0 {
+            return weather.current.weather[0].icon
+        }
+        return "sun.max.fill"
+    }
+    
+    func getTemperature(temp: Double) -> String {
+        return String(format: "%0.1f", temp)
+    }
+    
+    var temperature: String {
+        return getTemperature(temp: weather.current.temp)
+    }
+    
+    var conditions: String {
+        if weather.current.weather.count > 0 {
+            return weather.current.weather[0].main
+        }
+        return ""
```

Comment:
You can use `weather.current.weather.first?.main ?? ""` here

Reply:
Fixed.)